client/src/js/components/bhDateEditor.js

```diff
@@ -0,0 +1,38 @@
+/**
+* The date editor component
+*
+* @module components/dateEditor
+*
+* @description a component to deal with date, it let you choose a date
+* This component can do his own validation based on parameters values provided*
+* 
+**/
+
+//A date editor component controller
+
+function dateEditorController (){
+	var ctrl = this;
+
+	ctrl.editMode = false;
+	ctrl.dateValue =  ctrl.dateValue || new Date();
+
+	function changeMode (){
+		ctrl.editMode = !ctrl.editMode;
+	}
+
+	ctrl.changeMode = changeMode;
+}
+
+//component implementation
+angular.module('bhima.components').component('bhDateEditor', {
+	templateUrl : '/partials/templates/bhDateEditor.tmpl.html',
+	controller : dateEditorController,
+	bindings : {
+		dateValue : '=',
+		dateFormat : '<',
+		minDate : '<',
+		maxDate : '<',
+		form : '<',
```

Comment:
Instead of passing in the parent form, why not use a child form to perform local form validation?  The `bhCurrencyInput` also makes this mistake.  However, the [bhCurrencyComponent](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/partials/templates/bhFindPatient.tmpl.html#L6) correctly uses a child form for local form validation.